src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -319,6 +322,57 @@ CreateReferenceTableColocationId()
 
 
 /*
+ * DeleteAllReferenceTablePlacementsFromNode function iterates over list of reference
+ * tables and deletes all reference table placements from pg_dist_shard_placement table
+ * for given worker node. It also modifies replication factor of the colocation group of
+ * reference tables so that replication factor will be equal to number of workers.
+ */
+void
+DeleteAllReferenceTablePlacementsFromNode(char *workerName, uint32 workerPort)
+{
+	List *referenceTableList = ReferenceTableOidList();
+	ListCell *referenceTableCell = NULL;
+
+	Oid firstReferenceTableId = InvalidOid;
+	uint32 referenceTableColocationId = INVALID_COLOCATION_ID;
+
+	List *workerNodeList = NIL;
+	int workerCount = 0;
+
+	/* if there is no reference table, we do not need to do anything */
+	if (list_length(referenceTableList) == 0)
+	{
+		return;
+	}
+
+	referenceTableList = SortList(referenceTableList, CompareOids);
```

Comment:
Maybe a single line comment why are we sorting the table list? Is that to prevent deadlocks among concurrent `remove_node()` calls?